lib/mongo.ex

```diff
@@ -185,6 +185,21 @@ defmodule Mongo do
   end
 
   @doc """
+  Selects a single document in a collection and return the document or null
+  """
+  @spec findOne(Pool.t, collection, BSON.document, Keyword.t) :: BSON.document
+  def findOne(pool, coll, filter, opts \\ []) do
+    opts = Dict.put(opts, :limit, 1)
+    cursor = find(pool, coll, filter, opts)
```

Comment:
Call the `Connection` module directly instead of going through a cursor when we only need one result. `Connection` already has a `find_one` function so it should just be a matter of converting some options and then calling the function.